File: app/src/main/java/com/github/malitsplus/shizurunotes/common/BindingAdapter.java
package com.github.malitsplus.shizurunotes.common;

import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.constraintlayout.widget.ConstraintLayout;

import com.allen.library.SuperTextView;
import com.squareup.picasso.Picasso;

public class BindingAdapter {

    @androidx.databinding.BindingAdapter(value = {"bind:imageUrl", "bind:placeHolder"}, requireAll = false)
    public static void loadImage(ImageView view, String imageUrl, Integer placeHolder) {
        if (placeHolder != null){
            Picasso.get()
                    .load(imageUrl)
                    .placeholder(placeHolder)
                    .into(view);
        }else {
            Picasso.get()
                    .load(imageUrl)
                    .into(view);
        }
    }

    @androidx.databinding.BindingAdapter("bind:src")
    public static void loadSrc(ImageView view, Integer src) {
        view.setImageResource(src);
    }

    @androidx.databinding.BindingAdapter("bind:sRightText")
    public static void setRightString(SuperTextView view, String src) {
        view.setRightString(src);
    }
}

File: app/src/main/java/com/github/malitsplus/shizurunotes/ui/base/BindingAdapter.kt
package com.github.malitsplus.shizurunotes.ui.base

import android.util.TypedValue
import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.allen.library.SuperTextView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.github.malitsplus.shizurunotes.data.Skill

@BindingAdapter(value = ["imageUrl", "placeHolder", "errorHolder"], requireAll = false)
fun loadImage(view: ImageView, imageUrl: String?, placeHolder: Int?, errorHolder: Int?) {
    when{
        !imageUrl.isNullOrEmpty() && placeHolder != null && errorHolder != null ->
            Glide.with(view.context)
                .load(imageUrl)
                .placeholder(placeHolder)
                .error(errorHolder)
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(view)
        !imageUrl.isNullOrEmpty() && placeHolder != null ->
            Glide.with(view.context)
                .load(imageUrl)
                .placeholder(placeHolder)
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(view)
        !imageUrl.isNullOrEmpty() && errorHolder != null ->
            Glide.with(view.context)
                .load(imageUrl)
                .error(errorHolder)
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(view)
        !imageUrl.isNullOrEmpty() ->
            Glide.with(view.context)
                .load(imageUrl)
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(view)
    }
}

@BindingAdapter("src")
fun loadSrc(view: ImageView, src: Int) {
    view.setImageResource(src)
}

@BindingAdapter("sRightText")
fun setRightString(view: SuperTextView, src: String) {
    view.setRightString(src)
}

@BindingAdapter("sLeftText")
fun setLeftString(view: SuperTextView, src: String) {
    view.setLeftString(src)
}

@BindingAdapter("checkSummonSkill")
fun setVisibility(view: View, skill: Skill?) {
    view.visibility = if (skill != null && skill.enemyMinionList.isNotEmpty()) {
        View.VISIBLE
    } else {
        View.GONE
    }
}